app/models/spree/address.rb

```diff
@@ -15,7 +15,9 @@ class Address < ApplicationRecord
 
     validates :address1, :city, :country, :phone, presence: true
     validates :company, presence: true, unless: -> { first_name.blank? || last_name.blank? }
-    validates :firstname, :lastname, presence: true, unless: -> { company.present? }
+    validates :firstname, :lastname, presence: true, if: -> do
+      company.blank? || company == I18n.t(:unused)
```

Comment:
Hmm... I don't really like this hack where we set various fields to be `unused` in general... I'd really like it if we completely removed that at some point. But maybe that's an issue for another day.

Assuming we're sticking with that for now; we should be comparing `company` to `'unused'` here instead of `I18n.t(:unused)`, because the relevant lines are here:

https://github.com/openfoodfoundation/openfoodnetwork/blob/1b79fdd333f738de2d76bd215f4c0c3efd205897/app/models/concerns/set_unused_address_fields.rb#L13-L14

Reply:
```suggestion
      company.blank? || company == 'unused'
```

Reply:
You're right, I've created an issue: #8697